src/DripsDeployer.sol

```diff
@@ -0,0 +1,412 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.17;
+
+import {AddressDriver} from "./AddressDriver.sol";
+import {Caller} from "./Caller.sol";
+import {Drips} from "./Drips.sol";
+import {ImmutableSplitsDriver} from "./ImmutableSplitsDriver.sol";
+import {Managed, ManagedProxy} from "./Managed.sol";
+import {NFTDriver} from "./NFTDriver.sol";
+import {OperatorInterface, RepoDriver} from "./RepoDriver.sol";
+import {Ownable2Step} from "openzeppelin-contracts/access/Ownable2Step.sol";
+import {Address} from "openzeppelin-contracts/utils/Address.sol";
+import {Create2} from "openzeppelin-contracts/utils/Create2.sol";
+
+struct Module {
+    bytes32 salt;
+    uint256 amount;
+    bytes initCode;
+}
+
+contract DripsDeployer is Ownable2Step {
+    // slither-disable-next-line naming-convention
+    bytes32[] internal _moduleSalts;
+    address public immutable initialOwner;
+
+    function args() public view returns (bytes memory) {
+        return abi.encode(initialOwner);
+    }
+
+    constructor(address initialOwner_) {
+        // slither-disable-next-line missing-zero-check
+        initialOwner = initialOwner_;
+        _transferOwnership(initialOwner);
+    }
+
+    function deployModules(
+        Module[] calldata modules1,
+        Module[] calldata modules2,
+        Module[] calldata modules3,
+        Module[] calldata modules4
+    ) public onlyOwner {
+        _deployModules(modules1);
+        _deployModules(modules2);
+        _deployModules(modules3);
+        _deployModules(modules4);
+    }
+
+    function _deployModules(Module[] calldata modules) internal {
+        for (uint256 i = 0; i < modules.length; i++) {
+            Module calldata module = modules[i];
+            _moduleSalts.push(module.salt);
+            // slither-disable-next-line reentrancy-eth,reentrancy-no-eth
+            Create3.deploy(module.amount, module.salt, module.initCode);
+        }
+    }
+
+    function moduleSalts() public view returns (bytes32[] memory) {
+        return _moduleSalts;
+    }
+
+    function moduleAddress(bytes32 salt) public view returns (address addr) {
+        return Create3.computeAddress(salt);
+    }
+}
+
+abstract contract BaseModule {
+    address public immutable dripsDeployer;
+    bytes32 public immutable moduleSalt;
+
+    constructor(address dripsDeployer_, bytes32 moduleSalt_) {
+        require(
+            address(this) == Create3.computeAddress(moduleSalt_, dripsDeployer_),
+            "Invalid module deployment salt"
+        );
+        dripsDeployer = dripsDeployer_;
+        moduleSalt = moduleSalt_;
+    }
+
+    function args() public view virtual returns (bytes memory);
```

Comment:
What is the point of `args()`? It is currently not used. Is the purpose just on-chain documentation of used parameters for actual deployment? 

I get the point of `deploymentArgs` and logicArgs` which are used.

Reply:
All the `...args` functions are used for contract verification. I don't quite understand why, but Etherscan requires filing not only the source code, but also the constructor arguments, for whatever reason it can't extract them from the initcode and the constructor source code. We can patch together the arguments in the deployment script or we can do it in the contract. Doing this in the contract is closer to the source of truth, and it also nicely documents the deployment on-chain.